apps/frontend/src/components/profile/pill/expansion.tsx

```diff
@@ -25,6 +27,15 @@ export const PillCollectionExpansion: FC<PillCollectionExpansionProps> = (
 	const { editable, user, attributes, session } = props;
 	const [expanded, setExpanded] = useState(false);
 
+	const kinks = useAttributeList("kink");
+	const activeKinkIds = useMemo(
+		() =>
+			filterBy(session.user.profile.attributes, "type", "kink")
+				.map(({ id }) => kinks.find((kink) => kink.id === id)?.metadata.pair)
```

Comment:
doing a find in a map is a little messy, alternatively you could `reduce` kinks into an object where keys correspond to ids and values correspond to `metadata.pair`.

```js
const kinkPairs = kinks.reduce((pairs, kink) => {
  pairs[kink.id] = kink.metadata.pair;
  return pairs;
}, {});
```